src/Workspaces/Remote/Core/Services/AssetSource.cs

```diff
@@ -13,21 +13,14 @@ namespace Microsoft.CodeAnalysis.Remote
     internal abstract class AssetSource
     {
         private readonly AssetStorage _assetStorage;
-        private readonly int _scopeId;
 
-        protected AssetSource(AssetStorage assetStorage, int scopeId)
+        protected AssetSource(AssetStorage assetStorage)
         {
             _assetStorage = assetStorage;
-            _scopeId = scopeId;
 
-            _assetStorage.RegisterAssetSource(_scopeId, this);
+            _assetStorage.SetAssetSource(this);
```

Comment:
❓ Does this mean if we create an `AssetSource` while an unrelated OOP request is in process, it could cause the unrelated request to fail?